tests/test_target_storage_types.py

```diff
@@ -393,183 +430,233 @@ def test17_target_storage_format_multiple_packages_from_same_product_package(sel
         self.validate_move_lines(apple_lines_done2, self.apple, 2, 2, is_package=True, package_name=package2.name)
 
     def test18_target_storage_format_multiple_packages_from_same_product_pallet_products(self):
-        """Tests that mutiple sets of the same product can be made from a single picking
-           for target_storage_format pallet_products.
+        """Tests that mutiple sets of the same product
+           can be made from a single picking for
+           target_storage_format pallet_products.
         """
         # Two packages from same product
         Package = self.env['stock.quant.package']
         package1 = Package.get_package('test_package1', create=True)
         package2 = Package.get_package('test_package2', create=True)
 
-        create_info = [{'product': self.apple, 'qty': 4}]
+        create_info = [{'product': self.apple, 'qty': 5}]
         picking = self.create_picking(self.picking_type_in,
                                       products_info=create_info,
                                       confirm=True)
-        products_info = [{'product_barcode': self.apple.barcode, 'qty': 2}]
-        self.validate_move_lines(picking.move_line_ids, self.apple, 4, 0)
+        products_info1 = [{'product_barcode': self.apple.barcode, 'qty': 2}]
+        products_info2 = [{'product_barcode': self.apple.barcode, 'qty': 3}]
+        self.validate_move_lines(picking.move_line_ids, self.apple, 5, 0)
 
         self.picking_type_in.u_target_storage_format = 'pallet_products'
-        picking.update_picking(products_info=products_info, result_package_name=package1.name)
+        picking.update_picking(products_info=products_info1, result_package_name=package1.name)
         apple_lines_done = picking.move_line_ids.filtered(lambda x: x.product_id == self.apple and x.qty_done > 0)
         self.validate_move_lines(apple_lines_done, self.apple, 2, 2, is_pallet=True, package_name=package1.name)
 
-        picking.update_picking(products_info=products_info, result_package_name=package2.name)
+        picking.update_picking(products_info=products_info2, result_package_name=package2.name)
         apple_lines_done2 = picking.move_line_ids.filtered(lambda x: x.product_id == self.apple and x.qty_done > 0)
         apple_lines_done2 -= apple_lines_done
-        self.validate_move_lines(apple_lines_done2, self.apple, 2, 2, is_pallet=True, package_name=package2.name)
+        self.validate_move_lines(apple_lines_done2, self.apple, 3, 3, is_pallet=True, package_name=package2.name)
 
     def test19_target_storage_format_multiple_packages_from_same_product_pallet_packages(self):
-        """Tests that mutiple sets of the two products can be made from a single picking
+        """Tests that mutiple sets of the two products
+           can be made from a single picking
+           for target_storage_format pallet_packages.
```

Comment:
Can we add something like: two packages inside one pallet, and in the next test two pallets and one package per pallet?

Reply:
And add some asserts to check that this has happened? like apple_lines_done.u_result_parent_package_id == apple_lines_done2.u_result_parent_package_id, and != for the other test?

Reply:
in some ways the package_name in validate_move_lines already the comparison does this as both are passed the same (or different) names on validation